### PR TITLE
Feature/mds 2900 remove duplicate variances from db

### PR DESCRIPTION
**Pull request by mhuseinov**

### CODE REVIEW

File: services/core-web/common/actionCreators/incidentActionCreator.js [hunk 1]
```diff
@@ -69,3 +69,22 @@ export const fetchIncidents = (payload) => (dispatch) => {
     .catch(() => dispatch(error(reducerTypes.GET_INCIDENTS)))
     .finally(() => dispatch(hideLoading()));
 };
+
+export const deleteMineIncident = (mineGuid, incidentGuid) => (dispatch) => {
+  dispatch(request(reducerTypes.DELETE_INCIDENT));
+  return CustomAxios()
```
Comment (mmumby) [inline services/core-web/common/actionCreators/incidentActionCreator.js:75]: add `dispatch(showLoading());` after line 74 to all actionCreators that have a `.finally(() => dispatch(hideLoading()));`

File: services/core-web/common/actionCreators/permitActionCreator.js [hunk 2]
```diff
@@ -133,3 +133,38 @@ export const removePermitAmendmentDocument = (
     .catch(() => dispatch(error(reducerTypes.UPDATE_PERMIT_AMENDMENT_DOCUMENT)))
     .finally(() => dispatch(hideLoading()));
 };
+
+export const deletePermit = (mineGuid, permitGuid) => (dispatch) => {
+  dispatch(request(reducerTypes.DELETE_PERMIT));
+  return CustomAxios()
+    .delete(`${ENVIRONMENT.apiUrl}${API.PERMITDELETE(mineGuid, permitGuid)}`, createRequestHeader())
+    .then((response) => {
```
Comment (mmumby) [inline services/core-web/common/actionCreators/permitActionCreator.js:141]: update casing of 'PERMITDELETE` to be `PERMIT_DELETE` same goes for `PERMITAMENDMENT` below

File: services/core-web/src/components/mine/Permit/MinePermitTable.js [hunk 3]
```diff
@@ -179,26 +179,55 @@ const columns = [
...
-          isMajorMine={text.major_mine_ind}
+
+      const deletePermitPopUp = (
+        <Popconfirm
+          placement="topLeft"
+          {...(() => {
+            return record.permit.bonds && record.permit.bonds.length > 0
+              ? {
+                  title: "You can not delete permit with bonds",
+                  okText: "Ok",
```
Comment (mmumby) [inline services/core-web/src/components/mine/Permit/MinePermitTable.js:190]: "You cannot delete a permit that has associated bond records"

File: services/core-web/common/actionCreators/permitActionCreator.js (hunk 2 above)
Comment (mhuseinov) [reply services/core-web/common/actionCreators/permitActionCreator.js:141]: For consistency I followed the pattern of 'Permit' in API.js:
PERMITSTATUSCODES
PERMITAMENDMENTS
PERMITAMENDMENTDOCUMENT .....
Do we want to update all of them?

Comment (mmumby) [reply services/core-web/common/actionCreators/permitActionCreator.js:141]: yeah, they should all be snake_cased, They must have been overlooked.

File: services/core-web/common/actionCreators/incidentActionCreator.js (hunk 1 above)
Comment (mhuseinov) [reply services/core-web/common/actionCreators/incidentActionCreator.js:76]: done

File: services/core-web/src/components/mine/Permit/MinePermitTable.js (hunk 3 above)
Comment (mhuseinov) [reply services/core-web/src/components/mine/Permit/MinePermitTable.js:191]: done

**Pull request closed without merging**